src/ControlCodeParser.php

```diff
@@ -171,6 +175,12 @@ public function handleData($data)
                         break;
                     }
                 }
+            } else if ($type === 'ss2' || $type === 'ss3') {
+                $data = substr($this->buffer, 0, 3);
```

Comment:
Logic looks about right! :+1: This needs an additional check to ensure that 3 bytes are available in the buffer and set `$found` accordingly (also needs additional tests).